### ISSUE TITLE
L16/48000 stereo/mono support

### ISSUE THREAD

**Issue by premultiply** [alfredh/baresip#684]
Hi,
playing around with serveral other SIP UAs I wonder why baresip has no support for Linear 16 Bit Audio at 48000 Hz sampling rate. Or even other bitdepths like 20 or 24 bit.
Is there any technical reason for this or was it just not added to the list of formats?

**Comment by alfredh** [comment]
here is the list of codecs from the l16 module:

```c
/* See RFC 3551 */
static struct aucodec l16v[NR_CODECS] = {
{LE_INIT, "10", "L16", 44100, 44100, 2, 2, 0, 0, encode, 0, decode, 0, 0, 0},
{LE_INIT,    0, "L16", 32000, 32000, 2, 2, 0, 0, encode, 0, decode, 0, 0, 0},
{LE_INIT,    0, "L16", 16000, 16000, 2, 2, 0, 0, encode, 0, decode, 0, 0, 0},
{LE_INIT,    0, "L16",  8000,  8000, 2, 2, 0, 0, encode, 0, decode, 0, 0, 0},
{LE_INIT, "11", "L16", 44100, 44100, 1, 1, 0, 0, encode, 0, decode, 0, 0, 0},
{LE_INIT,    0, "L16", 32000, 32000, 1, 1, 0, 0, encode, 0, decode, 0, 0, 0},
{LE_INIT,    0, "L16", 16000, 16000, 1, 1, 0, 0, encode, 0, decode, 0, 0, 0},
{LE_INIT,    0, "L16",  8000,  8000, 1, 1, 0, 0, encode, 0, decode, 0, 0, 0},
};
```

48000 can easily be added to this table.

L24 can also be added but should be a new module -- `l24.so`

**Comment by premultiply** [comment]
Thanks Alfred!
I've added support for L16 48000 by https://github.com/alfredh/baresip/pull/685

**Comment by alfredh** [comment]
fixed in commit 96ebfe51f303010cfb255f5a8117ab9f74046790

thanks @premultiply
